src/main/scala/scoverage/ScoverageSbtPlugin.scala

```diff
@@ -329,29 +334,72 @@ object ScoverageSbtPlugin extends AutoPlugin {
   private def checkCoverage(
       coverage: Coverage,
       log: Logger,
-      min: Double,
+      min: CoverageMinima,
       failOnMin: Boolean
   ): Unit = {
+    val ok: Boolean = checkCoverage(coverage, "Total", log, min.total) &&
+      coverage.packages.forall(x =>
+        checkCoverage(x, s"Package:${x.name}", log, min.perPackage)
+      ) &&
+      coverage.files.forall(x =>
+        checkCoverage(x, s"File:${x.filename}", log, min.perFile)
+      )
 
-    val cper = coverage.statementCoveragePercent
-    val cfmt = coverage.statementCoverageFormatted
+    if (!ok && failOnMin)
+      throw new RuntimeException("Coverage minimum was not reached")
 
+    log.info(
+      s"All done. Coverage was [${coverage.statementCoverageFormatted}%]"
```

Comment:
So depending your thoughts up above, we may need to change this to not just output the statement total but an actual average total, or we might as well output both totals here.

Reply:
will do